internal/mode/static/manager.go

```diff
@@ -73,16 +80,118 @@ func StartManager(cfg config.Config) error {
 		return fmt.Errorf("cannot build runtime manager: %w", err)
 	}
 
+	hc := &healthChecker{}
+	if cfg.HealthConfig.Enabled {
+		if err := mgr.AddReadyzCheck("readyz", hc.readyCheck); err != nil {
+			return fmt.Errorf("error adding ready check: %w", err)
+		}
+	}
+
 	recorderName := fmt.Sprintf("nginx-kubernetes-gateway-%s", cfg.GatewayClassName)
 	recorder := mgr.GetEventRecorderFor(recorderName)
 	logLevelSetter := newZapLogLevelSetter(cfg.AtomicLevel)
 
+	ctx := ctlr.SetupSignalHandler()
```

Comment:
Majority of the changes following this point were simply pulling out a bunch of logic into the `registerControllers` function. The diff just makes it look worse than it really is.